### ISSUE TITLE
Predict soft segmentation masks

### ISSUE THREAD

**Issue by matteo-tafuro** [MIC-DKFZ/nnUNet#1302]
> **Disclaimer: I'm a complete newbie to the nnU-Net framework.**

I'm trying to predict _soft_ segmentation masks (i.e. continuous values between 0 and 1 for each class) using nnU-Net, rather than standard _hard_ masks. Does the current implementation support such an option? If not, would you mind giving some directions wrt the files that need changes? Not being familiar with the framework makes it really hard to navigate through the code and all the subdirectories.

Thanks!

**Comment by JackRio** [comment]
Highly recommend you to read the documentation in the readme

If you want to get the segmentation mask during validation check the `--npz` parameter under [Model Training](https://github.com/MIC-DKFZ/nnUNet#model-training)

During inference `--save_npz` saves the softmax output read up under [Run Inference](https://github.com/MIC-DKFZ/nnUNet#run-inference)

**Comment by lutai14** [comment]
Hey Jack, thanks for your reply. I see, thank you! On a similar note, does the framework support any loss function that accepts soft masks as labels?

**Comment by JackRio** [comment]
Not exactly sure what you want there. Can you elaborate what you are trying to do?

**Comment by FabianIsensee** [comment]
Hey, so for export @JackRio has got you covered. This is what you are looking for.
You can not train with soft labels though. This would require a LOT of rework
